ddl/ddl_api.go

```diff
@@ -351,6 +351,22 @@ func ResolveCharsetCollation(charsetOpts ...ast.CharsetOpt) (string, string, err
 	return chs, coll, nil
 }
 
+func OverwriteCollationWithBinaryFlag(colDef *ast.ColumnDef, chs, coll string) (newChs string, newColl string) {
+	ignoreBinFlag := colDef.Tp.Charset != "" && (colDef.Tp.Collate != "" || containsColumnOption(colDef, ast.ColumnOptionCollate))
```

Comment:
If there is an explicit collate clause, we shouldn't ignore the bin flag too.

Reply:
If both column charset and column collation is set, the bin flag is ignored. This is consistent with MySQL 8.0.

Reply:
```sql
select version();
create table t (a varchar(10) binary charset utf8 collate utf8_general_ci);
```

```console
+-----------+
| version() |
+-----------+
| 8.0.18    |
+-----------+
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (\n  `a` varchar(10) CHARACTER SET utf8 COLLATE utf8_general_ci DEFAULT NULL\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Reply:
```
mysql> create table t (a varchar(10) binary  collate utf8_general_ci);
mysql> show create table t;
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                             |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` varchar(10) CHARACTER SET utf8 COLLATE utf8_bin DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
It's too disgusting.

Reply:
I hope we can add a warning for `binary`. Users should not use it.

Reply:
I guess the warning should be thrown by the parser?